dali/operators/generic/one_hot.h

```diff
@@ -31,17 +31,24 @@ namespace detail {
 template<typename Out, typename In>
 void DoOneHot(kernels::OutTensorCPU<Out, DynamicDimensions> output,
               kernels::InTensorCPU<In, DynamicDimensions> input, int num_classes,
-              same_as_t<Out> on_value, same_as_t<Out> off_value, int ndims, int axis = 0) {
+              same_as_t<Out> on_value, same_as_t<Out> off_value, int axis) {
   auto in = input.data;
   auto out = output.data;
-  for (int i = 0; i < volume(output.shape); ++i) out[i] = off_value;
-  int64_t inner = volume(input.shape.last(ndims - axis));
-  for (int64_t i = 0, n = input.num_elements(); i < n; i++) {
-    int cls = in[i];
-    if (cls < 0 || cls >= num_classes)
-      continue;
-    int64_t out_idx = (i % inner) + cls * inner + (i / inner) * (inner * num_classes);
-    out[out_idx] = on_value;
+  axis = axis < 0 ? output.shape.sample_dim() - 1 : axis;
```

Comment:
Let the called do it (outside the sample loop)?

Reply:
done